Slim/Control/Queries.pm

```diff
@@ -4636,26 +4689,41 @@ sub worksQuery {
 		my $sth = $dbh->prepare_cached($sql);
 		$sth->execute( @{$p} );
 
-		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort );
-		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort);
+		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort, $covers, $album_ids);
+		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort, \$covers, \$album_ids);
 
 		while ( $sth->fetch ) {
-			#$id += 0;
+
+			my @allImages = split(/,/,$covers);
+			my $image = @allImages[0];
+			my $images = scalar @allImages > 4 ? join(',', @allImages[0..3]) : join(',',@allImages);
```

Comment:
Nitpick: only wrap the `@allImages` inside the condition, but keep the `join` outside, as we need it in any case.

The limitation to four images again seems to be a very specific UI decision: it's Material, and Material only. This doesn't belong here. Ship all or nothing, let the UI decide what to do about it. Or introduce another parameter to let the UI tell you how many images it wants. But no UI specific decision here, please.

Reply:
OK